sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Canonicalize.scala

```diff
@@ -85,6 +87,14 @@ object Canonicalize {
     case Not(GreaterThanOrEqual(l, r)) => LessThan(l, r)
     case Not(LessThanOrEqual(l, r)) => GreaterThan(l, r)
 
+    // order the list in the In operator
+    // we can do this only if all the elements in the list are literals with the same datatype
+    case i @ In(value, list)
```

Comment:
Can't we just reorder elements in list by `hashCode`?

Reply:
nice idea, thanks. I am doing it. It will allow also to cover more cases than simple literals. Thank you.